src/Watch.cpp

```diff
@@ -308,8 +308,158 @@ void Watch::parseWO(WatchInfo& wi, int first, int count) {
     wo.calories+= *it++ << 24;
 
     it = cb.memory.begin() + 64;
+
+    // For debugging, dump the data from the laps.  Perhaps this should only be enabled via --verbose?
+    std::cout << "Lap";
+    for (int i = 0; i < 16; i++) {
+        std::cout << std::setw(3) << i + 1 << " ";
+    }
+    std::cout << std::endl;
+    for (unsigned int lap = 0; lap < wo.lapcount; lap++) {
+        std::cout << lap + 1 << ": ";
+        for (int i = 0; i < 16; i++) {
+            std::cout << std::setw(3) << (int) *it++ << " ";
+        }
+        std::cout << std::endl;
+    }
+    std::cout << std::endl;
+
+    // Reset the it pointer back to the start of the lap data
+    it = cb.memory.begin() + 64;
     
-    //now every 16bytes is a laptime 1=hour 2=minutes 3=seconds 4+5=microseconds 10+9 lap distance 13=lapspeed in km/h
+    // Add a header to the lap data output
+    std::cout << wo.start_time.format() << std::endl;
+
+    std::cout << "Lap Absolutie Time            WorkoutTime Lap         HR  Dist  Spd       Pace" << std::endl;
+    std::cout << "--- ------------------------- ----------- ----------- --- ----- --------- --------" << std::endl;
+
+    //now every 16 bytes is a laptime
+    //1=hour 2=minutes 3=seconds 4=microseconds to be displayed as hex
+    //5=average hr 6-8=blank? 9-12=lap distance 13-16=lapspeed in 100m/h
+    wo.lapinfo = new LapInfo[wo.lapcount];
```

Comment:
As I said, it's was probably last century when I wrote C++ last.  I was honestly expecting to get a SIGSEGV for not freeing that memory and when I didn't I promptly forgot about it.  Too much Java lately I guess.

I'll check out std:vector and reimplement.